realm/realm-library/src/objectServer/java/io/realm/SyncConfiguration.java

```diff
@@ -182,6 +196,55 @@ public static RealmConfiguration forRecovery(String canonicalPath, @Nullable byt
         return forRecovery(canonicalPath, encryptionKey, schemaMediator);
     }
 
+    /**
+     * FIXME
+     *
+     * @param user
+     * @param partitionValue
+     * @return
+     */
+    @Beta
+    public static SyncConfiguration defaultConfig(RealmUser user, String partitionValue) {
```

Comment:
Any reason for all these config methods not being on `RealmUser`?

Reply:
Yes and no.... The original design intent was to keep the same pattern for local and synced Realms, so:

```
val localConfig = RealmConfiguration.Builder()....build()
val localRealm = Realm.getInstance(localConfig)

val syncConfig = SyncConfiguration.Builder(user, "partitionValue")....build()
val syncConfig = Realm.getInstance(syncConfig)
```

That doesn't really explain why these helper methods for creating the simple config are here though and if we do this, we should probably also add `RealmConfiguration.defaultConfig()` for symmetry.

Would it make sense to instead have?

```
val user = app.login(RealmCredentials.anonymous())
val syncConfig = user.defaultSyncConfiguration("partitionValue")
```

It might, but we wanted to move people away from having to go through users for configurations, which the above doesn't really do.

For this PR, I would like to keep them where they are...but it is definitely an area of the API we need to keep a close eye one.